src/System.Collections.Concurrent/tests/BlockingCollectionTests.cs

```diff
@@ -48,5 +48,1967 @@ public static void TestBlockingCollectionBasic()
+
+            //creates 8 consumers, each calling take to block itself
+            for (int i = 0; i < count; i++)
+            {
+                int myi = i;
+                Task.Run(() =>
+                {
+                    bc.Take();
+                    cde.Signal();
+                });
+            }
+            //create 8 producers, each calling add to unblock a consumer
+            for (int i = 0; i < count; i++)
+            {
+                int myi = i;
+                Task.Run(() =>
+                {
+                    bc.Add(new object());
+                });
+            }
+
+            //CountdownEvent waits till all consumers are unblocked
+            cde.Wait();
+            bc.CompleteAdding();
+        }
+
+        // This test shows how to cleanup of linkedCTS must be carefully handled to prevent 
+        // users of the source CTS mistakenly calling methods on disposed targets.
+        [Fact]
+        [OuterLoop]
+        public static void RunBlockingCollectionTest_Bug626345()
+        {
+            const int noOfProducers = 1;
+            const int noOfConsumers = 50;
+            const int noOfItemsToProduce = 2;
+            //Console.WriteLine("Producer: {0}, Consumer: {1}, Items: {2}", noOfProducers, noOfConsumers, noOfItemsToProduce);
+
+            BlockingCollection<long> m_BlockingQueueUnderTest = new BlockingCollection<long>(new ConcurrentQueue<long>());
+
+            Task[] producers = new Task[noOfProducers];
+            for (int prodIndex = 0; prodIndex < noOfProducers; prodIndex++)
+            {
+                producers[prodIndex] = Task.Run(() =>
+                {
+                    for (int dummyItem = 0; dummyItem < noOfItemsToProduce; dummyItem++)
+                    {
+                        int i = 0;
+                        for (int j = 0; j < 5; j++)
+                        {
+                            i += j;
+                        }
+                        m_BlockingQueueUnderTest.Add(dummyItem);
+
+                    }
+                });
+            }
+
+            //consumers
+            Task[] consumers = new Task[noOfConsumers];
+            for (int consumerIndex = 0; consumerIndex < noOfConsumers; consumerIndex++)
+            {
+                consumers[consumerIndex] = Task.Run(() =>
+                {
+                    while (!m_BlockingQueueUnderTest.IsCompleted)
+                    {
+                        long item;
+                        if (m_BlockingQueueUnderTest.TryTake(out item, 1))
+                        {
+                            int i = 0;
+                            for (int j = 0; j < 5; j++)
+                            {
+                                i += j;
+                            }
+                        }
+                    }
+                });
+            }
+
+            //Wait for the producers to finish.
+            //It is possible for some of the tasks in the array to be null, because the
```

Comment:
This comment looks inaccurate; I don't see how it's possible that any of the array elements are null, and if they are, the WaitAll would throw an exception.